sweep/txgenerator.go

```diff
@@ -125,10 +125,14 @@ func getPositiveYieldInputs(sweepableInputs []Input, maxInputs int,
 	for idx, input := range sweepableInputs {
 		// Can ignore error, because it has already been checked when
 		// calculating the yields.
-		size, _ := getInputWitnessSizeUpperBound(input)
+		size, isNestedP2SH, _ := getInputWitnessSizeUpperBound(input)
 
 		// Keep a running weight estimate of the input set.
-		weightEstimate.AddWitnessInput(size)
+		if isNestedP2SH {
+			weightEstimate.AddNestedP2WSHInput(size)
+		} else {
+			weightEstimate.AddWitnessInput(size)
+		}
```

Comment:
Arguably, we don't even need to check if it's a nested witness or not below since we're only seeking a relative ordering. Below we get a more precise estimate since it factors in things like the var int signal the number of inputs, the witness marker bytes, etc. 

Morphing the estimator to take a `sweep.Input` would cause an import cycle atm as the `sweep` package imports `lnwallet`. We could move `sweep.Input` into `lnwallet` but then that exacerbates the "catch all" trait of the `lnwallet` package as is. IMO, the package should be split up to 4+ (at the very least) distinct packages. 

An alternative approach would be to have the method in question take the input and the estimator, mutating the estimator, and returning no value. What do you think of that?

Reply:
With only segwit inputs, the relative ordering was correct, because all the common elements of (the serialization) of an input cancelled out. Now with nested p2wsh I think it is necessary to make this change.

Creating that helper method doesn't make it much better imo.

Splitting up in 4+ packages: how about going one step in that direction? Determine in which package you'd like to ultimately have `TxWeightEstimator` and only create that package in this PR. `TxWeightEstimator` has no dependencies on `lnwallet` is seems, so this should be an isolated change.

Reply:
I'd prefer to defer making the new package all together to avoid having to re-create/destroy the packages in the future. IMO in order to do it properly we need to take a step back, analyze dependancies like the one brought up here, and group the interfaces/structs/etc into logical components based on their responsibilities, context, etc.

Reply:
If the change to create a new package is small, then I think it's best to just do it right away. As @joostjager mentions, deferring it might make the change more significant at a later point.

Not sure what the change would be here though? Create a new package `sweep/TxWeightEstimator`, and move the `TxWeightEstimator` from `lnwallet` there. But this cannot take a `sweep.Input` (which would be quite nice!) since it imports `lnwallet`. Might be a bigger change to make this work..

Reply:
I couldn't resist giving it a try instead of writing another comment.
https://github.com/lightningnetwork/lnd/compare/master...joostjager:move-input

I created the package `lnwallet/input` (name not accurate) and moved in what needed to move. I think the end result is a logical group, but I have to admit that it involved many more renames than I expected.

I got to see only max 10 errors, so didn't now when it was finally over. But it builds.